rls/src/main/java/io/grpc/rls/RlsProtoConverters.java

```diff
@@ -188,9 +215,22 @@ static GrpcKeyBuilder convert(Map<String, ?> keyBuilder) {
       if (constantKeys == null) {
         constantKeys = ImmutableMap.of();
       }
+      checkUniqueKey(nameMatchers, constantKeys.keySet());
       return new GrpcKeyBuilder(names, nameMatchers, extraKeys, constantKeys);
     }
   }
 
+  private static void checkUniqueKey(List<NameMatcher> nameMatchers, Set<String> constantKeys) {
+    Set<String> keys = new HashSet<>(constantKeys);
+    List<String> extraKeys = ImmutableList.of("host", "service", "method");
```

Comment:
I wonder if there's a benefit of making this one a static class field, so we don't have to construct it on every call.